sdk/Pulumi/Deployment/IDeploymentBuilder.cs

```diff
@@ -0,0 +1,7 @@
+namespace Pulumi;
+
+internal interface IDeploymentBuilder
```

Comment:
I think its worth pulling this IDeploymentBuilder change into it's own PR and seeing if it makes sense to optionally expose it to non-service tests as well.